src/NuGet.Services.EndToEnd/ReadmeTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using NuGet.Services.EndToEnd.Support;
+using NuGet.Services.EndToEnd.Support.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NuGet.Services.EndToEnd
+{
+    [Collection(nameof(PushedPackagesCollection))]
+    public class ReadmeTests
+    {
+        private readonly PushedPackagesFixture _pushedPackages;
+        private readonly Clients _clients;
+        private readonly ITestOutputHelper _logger;
+
+        public ReadmeTests(PushedPackagesFixture pushedPackages, ITestOutputHelper logger)
+        {
+            _pushedPackages = pushedPackages;
+            _clients = pushedPackages.Clients;
+            _logger = logger;
+        }
+
+        [Fact]
+        public async Task PushedPackageWithReadmeIsAvailableInRegistration()
+        {
+            // Arrange
+            var package = await _pushedPackages.PrepareAsync(PackageType.EmbeddedReadmeFile, _logger);
+            var expectedPath = GetExpectedReadmePath(package);
+
+            // Act & Assert
+            var packageRegistrationList = await _clients.Registration.WaitForPackageAsync(package.Id, package.FullVersion, semVer2: false, logger: _logger);
+            Assert.All(packageRegistrationList, x => Assert.EndsWith(expectedPath, x.CatalogEntry.ReadmeUrl));
+        }
+
+        [Fact]
+        public async Task PushedPackageWithReadmeIsAvailableInFlatContainer()
+        {
+            // Arrange
+            var package = await _pushedPackages.PrepareAsync(PackageType.EmbeddedReadmeFile, _logger);
+            var expectedContent = GetReadmeData();
+            var expectedPath = GetExpectedReadmePath(package);
+
+            // Act & Assert
+            await _clients.FlatContainer.WaitForPackageAsync(package.Id, package.NormalizedVersion, _logger);
+            var readmeContents = await _clients.FlatContainer.TryAndGetFileStringContent(package.Id, package.NormalizedVersion, FlatContainerContentType.Readme, _logger);
+            Assert.All(readmeContents, x => Assert.Equal(expectedContent, x));
+        }
+
+        [Fact]
+        public async Task PushedPackageWithReadmeIsAvailableInSearch()

```

Comment:
>PushedPackageWithReadmeIsAvailableInSearch [](start = 26, length = 42)

The readme link doesn't exist in Search. Why is this test needed?

Reply:
Just want check if package is available in the search, I guess is not necessary here